lib/src/session.dart

```diff
@@ -271,6 +272,7 @@ enum SyncErrorCategory {
 /// These errors will terminate the network connection
 /// (disconnect all sessions  associated with the affected connection),
 /// and the error will be reported via the connection state change listeners of the affected sessions.
+@Deprecated("Handle SyncError or its subclasses instead of checking the error codes.")
```

Comment:
```suggestion
@Deprecated("Use SyncError or its subclasses instead.")
```

This repeats on many places. I suggest we use this generic deprecation text in all places